Components/Queues/PhpAmqpLibQueueManager.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues;
+
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\SerializerBuilder;
+use JMS\Serializer\SerializerInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageHandlerInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exchange\AMQPExchangeType;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\PhpAmqpLibDriver;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\Endpoint;
+use Smartbox\Integration\FrameworkBundle\Tools\Helper\SmartesbHelper;
+
+class PhpAmqpLibQueueManager implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+    use UsesSmartesbHelper;
+    use UsesExceptionHandlerTrait;
+
+    /**
+     * Represents the maximum size of a message
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 1;
+
+    // To use the default exchange pass an empty string
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Message Id
+     * @var int
+     */
+    private $id;
+
+    /**
+     * One or more AMQP connections. If one of the connection fail, the manager will try the next one.
+     * @var array AMQPStreamConnection
+     */
+    private $connections = [];
+
+    /**
+     * @var AMQPStreamConnection
+     */
+    private $connection;
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange
+     */
+    private $exchange;
+
+    /**
+     * @var \AMQPQueue
+     */
+    private $queue;
+
+    /**
+     * @var AMQPMessage
+     */
+    private $message;
+
+    /**
+     * Maximum number of messages to consume if param killAfter exists
+     * @var int
+     */
+    public $max;
+
+    /**
+     * PhpAmqpLibQueueManager constructor.
+     * @param \AMQPConnection[] $connections
+     */
+    public function __construct($connections = null, int $max = -1, string $format = 'json', SerializableInterface $serializer = null)
+    {
+        $this->connections = $connections;
+        $this->shoudlStop = false;
+        $this->max = $max;
+        $this->format = $format;
+        $this->serializer = $serializer ?? SerializerBuilder::create()->build();
```

Comment:
Part of what you have defined here it is also defined in the ```Components/Queues/Handler/PhpAmqpHandler.php``` like shouldStop, max, format, etc. Which class is the responsible of that?

Reply:
This class was removed.